### ISSUE TITLE
no-unused-expressions should catch property access

### ISSUE THREAD

**Issue by Daniel15** [eslint/eslint#5551]
With ESLint 2.3.0 and the `no-unused-expressions` rule enabled, this is lint-free (tested [on the demo](http://eslint.org/demo/)):
```javascript
var STUFF = {
  foo: 'bar'
};

STUFF.foo
```

Shouldn't the `STUFF.foo` be caught by `no-unused-expressions`, or even a totally separate rule? I understand that `STUFF.foo` could actually be a getter with side-effects and thus technically it's not unused, but I don't think that's very common in JavaScript at all.

The reason I noticed this was that I discovered code in our codebase that looked something like:
```javascript
function getDefaultFoo() {
  if (someCondition) {
    STUFF.foo;
  }
  ...
  return STUFF.bar;
}
```

Obviously `STUFF.foo` was meant to be `return STUFF.foo`.

**Comment by platinumazure** [comment]
My guess is we don't flag properties because the properties could be backed by getters or setters, which could have side effects.

**Comment by Daniel15** [comment]
I think it would still be good to have a way to optionally flag this as a lint warning too. Getters with side effects should be very rare.

**Comment by nzakas** [comment]
@Daniel15 I can't reproduce. I just verified that `STUFF.bar` is flagged with the latest version of ESLint. If you're using an older version, please try upgrading.

**Comment by Daniel15** [comment]
@nzakas Does the demo not use the latest version of ESLint? I can repro on the demo:

![screen shot 2016-03-11 at 2 13 16 pm](https://cloud.githubusercontent.com/assets/91933/13716998/799f7eb0-e793-11e5-9efe-146306f3b2c0.png)

Looks like we're using an older version internally (1.10.3), I'll see if I can upgrade it.

**Comment by nzakas** [comment]
The demo is using the latest version (2.4.0). Try clearing your cache.

I see that `STUFF.bar` is flagged in the demo as well.
